### PR TITLE
Add geoserver chart

### PR DESCRIPTION
**Pull request by LukasLohoff**
Introduces a helm chart for [`terrestris/docker-geoserver`](https://github.com/terrestris/docker-geoserver/)

Main configuration options:

* `environment.stableExtensions`: Comma-seperated list of stable geoserver extensions (`STABLE_EXTENSIONS` environment variable).
* `environment.javaOptions`: Custom java options (`EXTRA_JAVA_OPTS` environment variable).
* `extraEnv`: Map of additional environment variables passed to geoserver.
* `extraEnvFrom`: Pass additional environment variables from secrets, configmaps etc.
* `storage.datadir`: Where to mount the geoserver data dir.
* `storage.initDataUrl`: (Optional) External link to an compressed geoserver data dir (.tgz). The data will be extracted during startup.
* `persistence.enabled`: A default pvc (persistent volume claim) is used for persistent geoserver data
* `persistence.size`: Size of pvc (persistent volume claim)
* `persistence.useExisting`: Should an existing pvc (persistent volume claim) be used, default: `false`
* `persistence.existingPvcName`: The name of an existing pvc (persistent volume claim) that should be used to store geoserver data

@terrestris/devs Please review

### CODE REVIEW

File: charts/geoserver/templates/pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{{- if and .Values.persistence.enabled (not .Values.persistence.useExisting) -}}
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ include "mapproxy.fullname" . }}
```
Comment (buehner) [inline charts/geoserver/templates/pvc.yaml:6]: Is `mapproxy` intended here?

File: charts/geoserver/templates/pvc.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  labels:
+    {{- include "mapproxy.labels" . | nindent 4 }}
```
Comment (buehner) [inline charts/geoserver/templates/pvc.yaml:8]: Same here

File: charts/geoserver/templates/pvc.yaml (hunk 1 above)
Comment (LukasLohoff) [reply charts/geoserver/templates/pvc.yaml]: nope, good catch! fixed in 772e2359fa9d58989ed833893d320b4dda1ad6a0

File: charts/geoserver/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            - name: temp
+              mountPath: /mnt/download
+            - name: init-configmap
+              mountPath: /mnt/
+            - name: datadir
+              mountPath: {{ .Values.storage.datadir }}
+        {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
```
Comment (ahennr) [inline charts/geoserver/templates/statefulset.yaml:39]:
```suggestion
          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
```

Comment (ahennr) [reply charts/geoserver/templates/statefulset.yaml:39]: Herewith, the preferred version is the one defined in `Chart.yaml`

File: charts/geoserver/templates/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+              protocol: TCP
+          readinessProbe:
+            httpGet:
+              path: {{ .Values.probes.readinessUrl }}
+              port: http
+          livenessProbe:
+            httpGet:
+              path: {{ .Values.probes.livenessUrl }}
+              port: http
+              httpHeaders:
```
Comment (ahennr) [inline charts/geoserver/templates/statefulset.yaml:65]: Doesn't this depend on the configured livenessProbe URL?

File: charts/geoserver/values.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
+replicaCount: 1
+
+image:
+  repository: terrestris/geoserver
+  tag: 2.21.2
```
Comment (ahennr) [inline charts/geoserver/values.yaml:5]:
```suggestion
  tag: ""
```

File: charts/geoserver/values.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+service:
+  type: ClusterIP
+  port: 8080
+
+ingress:
+  enabled: false
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    kubernetes.io/tls-acme: "true"
+    cert-manager.io/cluster-issuer: letsencrypt
```
Comment (ahennr) [inline charts/geoserver/values.yaml:21]:
```suggestion
```

File: charts/geoserver/values.yaml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+service:
+  type: ClusterIP
+  port: 8080
+
+ingress:
+  enabled: false
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    kubernetes.io/tls-acme: "true"
```
Comment (ahennr) [inline charts/geoserver/values.yaml:20]:
```suggestion
    # kubernetes.io/tls-acme: "true"
```

**Review by ahennr — COMMENTED** [review]
Nice work @LukasLohoff 🔝

File: charts/geoserver/templates/statefulset.yaml (hunk 4 above)
Comment (LukasLohoff) [reply charts/geoserver/templates/statefulset.yaml]: yes, good point. I'll remove it to also allow other request

**Review by ahennr — APPROVED** [review]

**Review by buehner — APPROVED** [review]

**Comment by LukasLohoff** [comment]
Thanks for the reviews!

**Pull request merged**